module/os/linux/zfs/zfs_vfsops.c

```diff
@@ -1804,7 +1804,10 @@ zfs_resume_fs(zfsvfs_t *zfsvfs, dsl_dataset_t *ds)
 	objset_t *os;
 	VERIFY3P(ds->ds_owner, ==, zfsvfs);
 	VERIFY(dsl_dataset_long_held(ds));
+	dsl_pool_t *dp = spa_get_dsl(dsl_dataset_get_spa(ds));
+	dsl_pool_config_enter(dp, FTAG);
```

Comment:
Is it worth adding 
```
ASSERT(!dsl_pool_config_held(dp));
```
before dsl_pool_config_enter(..)?

Reply:
dsl_pool_config_enter() already does that, so I don't think that's needed.